packages/react-native-session-replay/ios/Sources/RCTTextViewRecorder.swift

```diff
@@ -0,0 +1,145 @@
+
+    internal let uiManager: RCTUIManager
+
+    internal init(uiManager: RCTUIManager) {
+        self.uiManager = uiManager
+    }
+
+    internal func extractTextFromSubViews(
+        subviews: [RCTShadowView]?
+    ) -> String? {
+        if let subviews = subviews {
+            return subviews.compactMap { subview in
+                if let sub = subview as? RCTRawTextShadowView {
+                    return sub.text
+                }
+                if let sub = subview as? RCTVirtualTextShadowView {
+                    // We recursively get all subviews for nested Text components
+                    return extractTextFromSubViews(subviews: sub.reactSubviews())
+                }
+                return nil
+            }.joined()
+        }
+        return nil
+    }
+
+    public func semantics(
+        of view: UIView,
+        with attributes: SessionReplayViewAttributes,
+        in context: SessionReplayViewTreeRecordingContext
+    ) -> SessionReplayNodeSemantics? {
+        guard let textView = view as? RCTTextView else {
+            return nil
+        }
+
+        var shadowView: RCTTextShadowView? = nil
+        let tag = textView.reactTag
+
+        RCTGetUIManagerQueue().sync {
+            shadowView = uiManager.shadowView(forReactTag: tag) as? RCTTextShadowView
+        }
+
+        if let shadow = shadowView {
+            // TODO: RUM-2173 check performance is ok
+            let text = extractTextFromSubViews(
+                subviews: shadow.reactSubviews()
+            )
+
+            let builder = RCTTextViewWireframesBuilder(
+                wireframeID: context.ids.nodeID(view: textView, nodeRecorder: self),
+                attributes: attributes,
+                text: text,
+                textAlignment: shadow.textAttributes.alignment,
+                textColor: shadow.textAttributes.foregroundColor?.cgColor,
+                textObfuscator: textObfuscator(context),
+                font: shadow.textAttributes.effectiveFont(), // Custom fonts are currently not supported for iOS
+                contentRect: shadow.contentFrame
+            )
+            let node = SessionReplayNode(viewAttributes: attributes, wireframesBuilder: builder)
+            return SessionReplaySpecificElement(subtreeStrategy: .ignore, nodes: [node])
+        }
+        return SessionReplayInvisibleElement.constant
+    }
+}
+
+internal struct RCTTextViewWireframesBuilder: SessionReplayNodeWireframesBuilder {
+    let wireframeID: WireframeID
+    /// Attributes of the base `UIView`.
+    let attributes: SessionReplayViewAttributes
+    /// The text.
+    let text: String?
+    /// The alignment of the text.
+    var textAlignment: NSTextAlignment
+    /// The color of the text.
+    let textColor: CGColor?
+    /// The text obfuscator.
+    let textObfuscator: SessionReplayTextObfuscating
+    /// The font used by the text field.
+    let font: UIFont?
+    /// The frame of the text content
+    let contentRect: CGRect
```

Comment:
Personal preference, but I'd say it's fine to drop these `///` they don't add much context.